lib/byron/bench/Restore.hs

```diff
@@ -0,0 +1,419 @@
+import Cardano.Wallet.Primitive.AddressDerivation
+    ( Depth (..)
+    , NetworkDiscriminant (..)
+    , PersistPrivateKey
+    , SomeMnemonic (..)
+    , WalletKey
+    , digest
+    , publicKey
+    )
+import Cardano.Wallet.Primitive.AddressDerivation.Shelley
+    ( ShelleyKey, generateKeyFromSeed )
+import Cardano.Wallet.Primitive.AddressDiscovery
+    ( IsOurs, IsOwned )
+import Cardano.Wallet.Primitive.AddressDiscovery.Sequential
+    ( SeqState, defaultAddressPoolGap, mkSeqStateFromRootXPrv )
+import Cardano.Wallet.Primitive.Model
+    ( currentTip, totalBalance, totalUTxO )
+import Cardano.Wallet.Primitive.Types
+    ( Address
+    , Block (..)
+    , BlockHeader (..)
+    , BlockchainParameters (..)
+    , ChimericAccount
+    , EpochNo (..)
+    , SlotId (..)
+    , SlotNo (..)
+    , StartTime (..)
+    , SyncProgress (..)
+    , UTxO (..)
+    , WalletId (..)
+    , WalletName (..)
+    , mkSyncTolerance
+    , slotParams
+    , syncProgressRelativeToTime
+    )
+import Cardano.Wallet.Unsafe
+    ( unsafeMkMnemonic, unsafeMkPercentage, unsafeRunExceptT )
+import Control.Concurrent
+    ( forkIO, threadDelay )
+import Control.DeepSeq
+    ( NFData, rnf )
+import Control.Exception
+    ( evaluate, throwIO )
+import Control.Monad
+    ( forM, mapM_, void )
+import Control.Monad.Fail
+    ( MonadFail )
+import Control.Monad.Trans.Except
+    ( runExceptT )
+import Criterion.Measurement
+    ( getTime, initializeTime, secs )
+import Data.Proxy
+    ( Proxy (..) )
+import Data.Quantity
+    ( Quantity (..) )
+import Data.Text
+    ( Text )
+import Data.Text.Class
+    ( ToText (..) )
+import Data.Time.Clock.POSIX
+    ( POSIXTime, getCurrentTime, getPOSIXTime, utcTimeToPOSIXSeconds )
+import Fmt
+    ( fmt, (+|), (+||), (|+), (||+) )
+import Ouroboros.Network.NodeToClient
+    ( NodeToClientVersionData (..) )
+import Say
+    ( sayErr )
+import System.Environment
+    ( getEnv )
+import System.FilePath
+    ( (</>) )
+import System.IO
+    ( BufferMode (..), hSetBuffering, stderr, stdout )
+import System.IO.Temp
+    ( withSystemTempFile )
+
+import qualified Cardano.BM.Configuration.Model as CM
+import qualified Cardano.BM.Data.BackendKind as CM
+import qualified Cardano.Wallet as W
+import qualified Cardano.Wallet.DB.Sqlite as Sqlite
+import qualified Data.Map.Strict as Map
+import qualified Data.Text as T
+
+
+-- | Run all available benchmarks. Can accept one argument that is a target
+-- network against which benchmarks below should be ran
+--
+-- (e.g. `--benchmark-arguments "mainnet"`)
+main :: IO ()
+main = do
+    hSetBuffering stdout NoBuffering
+    hSetBuffering stderr NoBuffering
+
+    (_logCfg, tr) <- initBenchmarkLogging Info
+    installSignalHandlers (return ())
+
+
+    -- TODO: We might want to support testnet
+    let network = Mainnet
+
+    -- Reading setup from env vars.
+    --
+    -- TODO: Look over whether this is sane or not.
+    configs <- getEnv "BYRON_CONFIGS"
+    nodeDB <- getEnv "NODE_DB"
```

Comment:
Maybe it would be easiest to set nodeDB to something like `./db` and socketPath to `./cardano-node.socket`. The logs will get dropped in `./logs` anyway. Then the script which executes the benchmark can take care of changing to the correct directory.

Reply:
I found it useful to be able to set NODE_DB when running locally, without nix.

`./cardano-node.socket` sounds sane though.